src/db.rs

```diff
@@ -218,39 +219,90 @@ impl DBStore {
         self.db.cf_handle(HEADERS_CF).expect("missing HEADERS_CF")
     }
 
-    pub(crate) fn iter_funding(&self, prefix: Row) -> impl Iterator<Item = Row> + '_ {
+    pub(crate) fn iter_funding(
+        &self,
+        prefix: HashPrefix,
+    ) -> impl Iterator<Item = SerializedHashPrefixRow> + '_ {
         self.iter_prefix_cf(self.funding_cf(), prefix)
     }
 
-    pub(crate) fn iter_spending(&self, prefix: Row) -> impl Iterator<Item = Row> + '_ {
+    pub(crate) fn iter_spending(
+        &self,
+        prefix: HashPrefix,
+    ) -> impl Iterator<Item = SerializedHashPrefixRow> + '_ {
         self.iter_prefix_cf(self.spending_cf(), prefix)
     }
 
-    pub(crate) fn iter_txid(&self, prefix: Row) -> impl Iterator<Item = Row> + '_ {
+    pub(crate) fn iter_txid(
+        &self,
+        prefix: HashPrefix,
+    ) -> impl Iterator<Item = SerializedHashPrefixRow> + '_ {
         self.iter_prefix_cf(self.txid_cf(), prefix)
     }
 
+    fn iter_cf<'a, const N: usize, F: FnMut(&[u8]) -> bool + 'a>(
+        &'a self,
+        cf: &rocksdb::ColumnFamily,
+        readopts: rocksdb::ReadOptions,
+        start: Option<&[u8]>,
+        mut filter_fn: F,
+    ) -> impl Iterator<Item = [u8; N]> + '_ {
+        let mut raw_iter = self.db.raw_iterator_cf_opt(cf, readopts);
+        let mut done = false;
+
+        if let Some(key) = start {
+            raw_iter.seek(key);
+        } else {
+            raw_iter.seek_to_first();
+        };
+
+        iter::from_fn(move || loop {
```

Comment:
I would suggest using the following https://github.com/romanz/electrs/commit/8845713acd5ec8d51430c6da944ce6b02fd966e4 - WDYT?

Reply:
Looks good apart from `self.raw.status()` being called after prefix mismatch and if already exhausted (`self.done`). That function is used for getting a potential error after `self.raw.key()` returned `None`. Fixed that in a new commit.

Reply:
Thanks!